tests/actions/orderEntryActions.test.js

```diff
@@ -61,14 +61,42 @@ describe ('Order entry configuration actions', () => {
             });
         });
 
-        it ('should dispatch `FETCH_ORDER_CONFIG_FAILURE` after an error', () => {
+        it ('should dispatch `FETCH_ORDER_CONFIG_FAILURE` if status is not 200', () => {
             moxios.wait(() => {
                 const request = moxios.requests.mostRecent();
                 request.respondWith({
+                    status: 204,
```

Comment:
You have a status of `204` but dispatching a failure.
Could you have a look at this resource for the 204 status code - [link](https://httpstatuses.com/204)

Reply:
The action creator throws an error when the status code is not equal to 200. So since 204 is for the successful request without content, I decided to use it in my tests.

Reply:
Okay.
If the expected action is an error, would it not be better to use a status code that more clearly shows this? 
`20*` status codes might not be the most appropriate in this situation.

Reply:
The expected action corresponds to a success code that is not 200. The error is thrown by the action, not the API.

Reply:
Just for more clarity.
A `204` status code means that there was not a failure in the action dispatch, rather it was successful but there is no content. This is quite different from what the `it` description in this test specifies.

Reply:
@flacode . I guess what @fuchodeveloper is saying is that a normal `axios` request will resolve a response with a status code of `204`. It won't be rejected and hence it won't go to the `catch` block. Your implementation might be right if you don't have a say over the status code coming from the API and all of that. But you should use a more precise status code if you can. 

And if you ask me, in this case (since I'm fetching data), I'll return a status `200` and an empty array and handle that in my `actionCreator`.

Reply:
@humanityjs, I agree with you. But how else would I test this async action? 
`return axiosInstance.get(`orderentryconfig?v=custom:(uuid,display)`)
    .then((response) => {
      if (response.status !== 200) {
        throw Error(response.statusText);
      }
      dispatch(loading('FETCH_ORDER_CONFIG', false));
      return response;
    })
    .then(response => dispatch(fetchConfigurationsSuccess(response.data)))
    .catch((error) => {
      dispatch(loading('FETCH_ORDER_CONFIG', false));
      dispatch(fetchConfigurationsFailure(error));
    });`